neurom/core/types.py

```diff
@@ -225,29 +225,25 @@ def _enum_accept_undefined(cls, value):
     # pylint: disable=protected-access
 
     # Use NeuriteType name
-    if isinstance(value, NeuriteType):
-        value_str = value.name
-        if value_str in cls._member_map_:
-            return cls._member_map_[value_str]
+    if isinstance(value, NeuriteType) and value.name in cls._member_map_:
+        return cls._member_map_[value.name]
 
     # Name given as string
-    elif isinstance(value, str):
-        if value in cls._member_map_:
-            return cls._member_map_[value]
+    if isinstance(value, str) and value in cls._member_map_:
+        return cls._member_map_[value]
```

Comment:
This is not the same as the previous version. In your version the value is tested multi times while we already know the result of the if statement.
For example, if a `NeuriteType` is given but the name is not in `cls._member_map_`, then now we also check that the value is a `str,` which we already know.

Reply:
Indeed. Revert!